config-linux.md

```diff
@@ -27,18 +27,22 @@ Namespaces are specified as an array of entries inside the `namespaces` root fie
 The following parameters can be specified to setup namespaces:
 
 * **`type`** *(string, required)* - namespace type. The following namespaces types are supported:
-    * **`pid`** processes inside the container will only be able to see other processes inside the same container
-    * **`network`** the container will have its own network stack
-    * **`mount`** the container will have an isolated mount table
-    * **`ipc`** processes inside the container will only be able to communicate to other processes inside the same container via system level IPC
-    * **`uts`** the container will be able to have its own hostname and domain name
-    * **`user`** the container will be able to remap user and group IDs from the host to local users and groups within the container
+    * **`pid`** processes inside the container will only be able to see other processes inside the same container. Support for this was added in Linux 2.6.24.
+    * **`network`** the container will have its own network stack. Support for this was added in Linux 2.6.24.
+    * **`mount`** the container will have an isolated mount table. Support for this was added in Linux 2.4.19.
+    * **`ipc`** processes inside the container will only be able to communicate to other processes inside the same container via system level IPC. Support for this was added in Linux 2.6.19.
+    * **`uts`** the container will be able to have its own hostname and domain name. Support for this was added in Linux 2.6.19.
+    * **`user`** the container will be able to remap user and group IDs from the host to local users and groups within the container. Support for this was added in Linux 3.8.
+    * **`cgroup`** the container will have an isolated view of its cgroup hierarchy. Support for this was added in Linux 4.6.
 
 * **`path`** *(string, optional)* - path to namespace file in the [runtime mount namespace](glossary.md#runtime-namespace)
 
 If a path is specified, that particular file is used to join that type of namespace.
 Also, when a path is specified, a runtime MUST assume that the setup for that particular namespace has already been done and error out if the config specifies anything else related to that namespace.
 
+If the runtime is unable to create or join ALL of the requested namespaces, it
+MUST fail.
```

Comment:
For `consistency with `runtime.md`][1], we should probably use “generate an error”.  Although @vishh's concern was [broader than namespaces][2], so I think we might want to put this language in step 2 of the [lifecycle][3].

Also, [one line per sentence][4] ;).

[1]: https://github.com/opencontainers/runtime-spec/blob/v0.5.0/runtime.md#errors
[2]: https://github.com/opencontainers/runtime-spec/pull/397#issuecomment-214505281
[3]: https://github.com/opencontainers/runtime-spec/blob/v0.5.0/runtime.md#lifecycle
[4]: https://github.com/opencontainers/runtime-spec/blob/v0.5.0/style.md#one-sentence-per-line